infra/blueprint-test/modules/gcloud/gcloud.go

```diff
@@ -0,0 +1,112 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package gcloud provides a set of helpers to interact with gcloud(Cloud SDK) binary
+package gcloud
+
+import (
+	"strings"
+
+	"github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/modules/utils"
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/shell"
+	"github.com/mitchellh/go-testing-interface"
+	"github.com/tidwall/gjson"
+)
+
+type Options struct {
+	gcloudBinary string         // path to gcloud binary
+	commonArgs   []string       // common arguments to pass to gcloud calls
+	logger       *logger.Logger // custom logger
+}
+
+type option func(*Options)
+
+func WithBinary(gcloudBinary string) option {
+	return func(f *Options) {
+		f.gcloudBinary = gcloudBinary
+	}
+}
+
+func WithCommonArgs(commonArgs []string) option {
```

Comment:
Should CommonArgs *override* the common args, or add to them?

I'm thinking this function might be more useful as:

```suggestion
func WithCommonArg(argument string, values ...string) option {
```

This way you can automatically inherit the default common args, but add to them if you want to without repeating the defaults.

Reply:
I think we should override them because we may not always want `--format json`. A simple usecase is if I want to alternate formatting like
https://github.com/terraform-google-modules/terraform-google-project-factory/blob/1bde0af3d3087dbd28c5b8bf296fccac094499e4/test/integration/shared_vpc_no_subnets/controls/gcloud.rb#L24